mailbox/backup/src/test/java/org/apache/james/mailbox/backup/DefaultMailboxBackupTest.java

```diff
@@ -0,0 +1,225 @@
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import org.apache.commons.compress.archivers.zip.ZipFile;
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.core.User;
+import org.apache.james.junit.TemporaryFolderExtension;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.backup.ZipAssert.EntryChecks;
+import org.apache.james.mailbox.backup.zip.ZipArchivesLoader;
+import org.apache.james.mailbox.backup.zip.Zipper;
+import org.apache.james.mailbox.extension.PreDeletionHook;
+import org.apache.james.mailbox.inmemory.MemoryMailboxManagerProvider;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.reactivestreams.Publisher;
+import org.testcontainers.shaded.org.apache.commons.io.IOUtils;
+
+import com.github.fge.lambdas.Throwing;
+import reactor.core.publisher.Mono;
+
+@ExtendWith(TemporaryFolderExtension.class)
+class DefaultMailboxBackupTest implements MailboxMessageFixture {
+
+    private static final String USER = "user";
+    private static final String OTHER_USER = "otherUser";
+
+    private static final User USER1 = User.fromUsername(USER);
+    private static final User USER2 = User.fromUsername(OTHER_USER);
+
+    private static final MailboxPath MAILBOX_PATH_USER1_MAILBOX1 = MailboxPath.forUser(USER, MAILBOX_1_NAME);
+    private static final MailboxPath MAILBOX_PATH_USER1_MAILBOX2 = MailboxPath.forUser(USER, MAILBOX_2_NAME);
+    private static final MailboxPath MAILBOX_PATH_OTHER_USER_MAILBOX1 = MailboxPath.forUser(OTHER_USER, MAILBOX_OTHER_USER_NAME);
+    private static final HashSet<PreDeletionHook> PRE_DELETION_HOOKS = new HashSet<>();
+
+    private final ArchiveService archiveService = new Zipper();
+    private final MailArchivesLoader archiveLoader = new ZipArchivesLoader();
+
+    private MailboxManager mailboxManager;
+    private BlobStore store;
+    private File destination;
+    private DefaultMailboxBackup backup;
+
+    @BeforeEach
+    void beforeEach(TemporaryFolderExtension.TemporaryFolder temporaryFolder) throws Exception {
+        destination = File.createTempFile("backup-test", ".zip", temporaryFolder.getTempDir());
+        mailboxManager = MemoryMailboxManagerProvider.provideMailboxManager(PRE_DELETION_HOOKS);
```

Comment:
Can't we use the newly introduced InMemoryIntegrationResource .defaultResources stuff, that is way better than this obscur factory method?

Reply:
I did'nt know it. Will have a look. Thanks

Reply:
I recently worked on this, so that definitly is a point of attention: spreading the <3 for this builder...